### PR TITLE
Add retry.sh

### PR DESCRIPTION
**Pull request by gerad**
**why**

It's been useful on agent-desktop and it'd be nice to get access to the
utility in earlier phases of the circle lifecycle.

**context**

https://github.com/sagansystems/agent-desktop/pull/3544

### CODE REVIEW

File: bin/circleci.sh [hunk 1]
```diff
@@ -13,4 +13,4 @@ fi
 git clone -b $BUILD_HARNESS_BRANCH $GITHUB_REPO
 make -C $BUILD_HARNESS_PROJECT deps circle:deps
 
-
+cat $BUILD_HARNESS_PROJECT/retry.sh >> ~/.bashrc
```
Comment (osterman) [inline bin/circleci.sh]: What are your thoughts about moving this to `circle:deps` target? ...to consolidate the places where dependencies are installed.

File: retry.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  local count=$retry_max
+  while [ $count -gt 0 ]; do
+    "$@" && break
+    count=$(($count - 1))
+    sleep 1
+  done
+
+  [ $count -eq 0 ] && {
+    echo "Retry failed [$retry_max]: $@" >&2
```
Comment (osterman) [inline retry.sh]: suggest outputting which attempt failed. e.g. `Retry failed [$count/$retry_max]: $@" >&2`

File: bin/circleci.sh (hunk 1 above)
Comment (gerad) [inline bin/circleci.sh]: 0f80be9

File: retry.sh (hunk 2 above)
Comment (gerad) [inline retry.sh]: 735f942

File: modules/Makefile.circleci [hunk 3]
```diff
@@ -31,6 +31,7 @@ circle\:deps:
      sudo chown ubuntu:ubuntu /usr/local/bin/jq; \
      sudo chmod 755 /usr/local/bin/jq; \
   fi
+	cat $(MAKEFILE_DIR)/retry.sh >> ~/.bashrc
```
Comment (osterman) [inline modules/Makefile.circleci]: nit: prefix the line with `@` to squelch output of command

**Comment by osterman** [comment]
aside from nit-pick, +2

File: modules/Makefile.circleci (hunk 3 above)
Comment (gerad) [inline modules/Makefile.circleci]: it was deliberate so you can see what's going on

**Pull request merged**